client/my-sites/domains/domain-management/name-servers/with-domain-nameservers.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import React, { useRef, useEffect } from 'react';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import useDomainNameserversQuery from 'calypso/data/domains/nameservers/use-domain-nameservers-query';
+import useUpdateNameserversMutation from 'calypso/data/domains/nameservers/use-update-nameservers-mutation';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+
+const useSuccessNotice = ( isSuccess ) => {
+	const showNotice = useRef();
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	useEffect( () => {
+		showNotice.current = () => {
+			dispatch(
+				successNotice( translate( 'Yay, the name servers have been successfully updated!' ), {
+					duration: 5000,
+					id: `nameserver-update-notification`,
+				} )
+			);
+		};
+	}, [ dispatch, translate ] );
+
+	useEffect( () => {
+		isSuccess && showNotice.current();
+	}, [ isSuccess ] );
+};
+
+const useErrorNotice = ( isError, error ) => {
```

Comment:
The notice hook approach looks a bit verbose. It would be even more obvious if we had 10+ noticess in here. I wonder if you've considered something different to handle notices.

Reply:
It's the way we handled those up until now but yup, here's another approach: https://github.com/Automattic/wp-calypso/pull/53665 – let me know if that makes better sense 👍🏻  /cc @jsnajdr